src/client/public/scripts/dash/dash.ts

```diff
@@ -253,67 +255,70 @@ function prTemplate(pr: api.PullRequest, newlyActionablePRs: string[]) {
     </div>`;
 }
 
-function hasNewActions(
-    newData: api.DashResponse, oldData: api.DashResponse): string[] {
-  if (!oldData) {
-    return [];
-  }
+function newActions(
+    newList: api.PullRequest[], oldList: api.PullRequest[]): string[] {
+  const result = [];
+  const oldActionablePRs: {[prUrl: string]: api.PullRequest} = {};
+  oldList.forEach((pr: api.PullRequest) => {
+    const {actionable} = statusToDisplay(pr);
+    if (!actionable) {
+      return;
+    }
 
-  const newlyActionablePRs: string[] = [];
-
-  const newActions =
-      (newList: api.PullRequest[], oldList: api.PullRequest[]) => {
-        const oldActionablePRs: {[prUrl: string]: api.PullRequest} = {};
-        oldList.forEach((pr: api.PullRequest) => {
-          const {actionable} = statusToDisplay(pr);
-          if (!actionable) {
-            return;
-          }
-
-          oldActionablePRs[pr.url] = pr;
-        });
-
-        for (const newPr of newList) {
-          const {actionable} = statusToDisplay(newPr);
-          if (!actionable) {
-            continue;
-          }
-
-          const oldPr = oldActionablePRs[newPr.url];
-          if (!oldPr) {
-            // New list has an actionable PR that didn't exist before
-            newlyActionablePRs.push(newPr.url);
-          }
-
-          if (JSON.stringify(newPr) !== JSON.stringify(oldPr)) {
-            newlyActionablePRs.push(newPr.url);
-          }
-        }
-      };
+    oldActionablePRs[pr.url] = pr;
+  });
 
-  newActions(newData.outgoingPrs, oldData.outgoingPrs);
-  newActions(newData.incomingPrs, oldData.incomingPrs);
+  for (const newPr of newList) {
+    const {actionable} = statusToDisplay(newPr);
+    if (!actionable) {
+      continue;
+    }
+
+    const oldPr = oldActionablePRs[newPr.url];
+    if (!oldPr) {
+      // New list has an actionable PR that didn't exist before
+      result.push(newPr.url);
+    }
 
-  return newlyActionablePRs;
+    if (JSON.stringify(newPr) !== JSON.stringify(oldPr)) {
+      result.push(newPr.url);
+    }
+  }
+  return result;
 }
 
-async function getDashData(userLogin: string|null) {
+async function fetchAndRender(userLogin: string|null): Promise<boolean> {
   // This allows you to see another users dashboard.
   const loginParams = userLogin ? `?login=${userLogin}` : '';
-  const res =
-      await fetch(`/api/dash.json${loginParams}`, {credentials: 'include'});
-  const newData = await res.json() as api.DashResponse;
-  const results = {
-    data: newData,
-    newActions: hasNewActions(newData, lastViewedData),
-  };
-
-  lastPolledData = newData;
+  // Send fetches in parallel.
+  const outgoing =
+      fetch(`/api/dash/outgoing${loginParams}`, {credentials: 'include'});
+  const incoming =
+      fetch(`/api/dash/incoming${loginParams}`, {credentials: 'include'});
+
+  // Render outgoing first.
+  const outgoingData =
+      await (await outgoing).json() as api.OutgoingDashResponse;
```

Comment:
Optional: Both incoming and outgoing look like that can be done in parallel 

```javascript
await Promise.all([
  getAndRenderOutgoing(),
  getAndRenderIncoming(),
]);
```
Main reasoning is it should clean up the outgoing and and incoming awaits.

Reply:
Done.